users/views.py

```diff
@@ -1,3 +1,69 @@
-from django.shortcuts import render
+import json, bcrypt, jwt
+
+from django.conf  import settings
+from django.views import View
+from django.http  import JsonResponse
+from django.forms import ValidationError
+
+from .models    import User
+from .validator import password_validator
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            user         = data['user']
+            password     = data['password']
+            address      = data['address']
+            phone_number = data['phone_number']
+
+            password_validator(password)
+            
+            if User.objects.filter(user=user).exists():
+                return JsonResponse({'message':'ID_ALREADY_EXISTS'}, status=400)
+            
+            if User.objects.filter(phone_number=phone_number).exists():
+                return JsonResponse({'message':'PHONE_NUMBER_ALREADY_EXISTS'}, status=400)
+
+            hashed_password = bcrypt.hashpw(password.encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
+            
+            User.objects.create(
+                user         = user,
+                password     = hashed_password,
+                address      = address,
+                phone_number = phone_number
+            )
+
+            return JsonResponse({'message':'SUCCESS'}, status=201)
+        except KeyError:
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+        except ValidationError as e:
+            return JsonResponse({'message':e.message}, status=400)
+
+class SignInView(View):
+    def post(self, request):
+        try: 
+            data = json.loads(request.body)
+            input_user     = data['user']
+            input_password = data['password']
+            confirm_user   = User.objects.get(user=input_user)
+
+            check_password = bcrypt.checkpw(input_password.encode('utf-8'), confirm_user.password.encode('utf-8'))      
+            if not check_password:
+                return JsonResponse({'message':'INVALID_PASSWORD'}, status=401)   
+            
+            access_token = jwt.encode({'user_id':confirm_user.id}, settings.SECRET_KEY, algorithm=settings.ALGORITHM)
+            return JsonResponse({
+                'message':'SUCCESS',
+                'token'  :access_token
+                }, status=200)
```

Comment:
1. line 55와 56 사이에 한 줄 띄워주세요!
2. "message" : "SUCCESS" 값은 필요한가요?
```suggestion
            access_token = jwt.encode({'user_id':confirm_user.id}, settings.SECRET_KEY, algorithm=settings.ALGORITHM)
            
            return JsonResponse({'message':'SUCCESS', 'token'  :access_token}, status=200)
```

Reply:
넵 수정하겠습니다

Reply:
조금 더 친절..(?)하기 위해 넣었는데, 없어도 무방함은 알고 있는지라.. 병민님 말씀하신 대로 삭제하겠습니다 ~!